lib/logging/entry.js

```diff
@@ -101,9 +101,16 @@ function Entry(resource, data) {
  * @return {module:logging/entity}
  */
 Entry.fromApiResponse_ = function(entry) {
-  // Only one of these are populated.
-  var data = entry.protoPayload || entry.jsonPayload || entry.textPayload;
-  return extend(new Entry(entry.resource, data), entry);
+  var data = entry[entry.payload];
+  var serializedEntry = extend(new Entry(entry.resource, data), entry);
+
+  if (serializedEntry.timestamp) {
+    var ms = serializedEntry.timestamp.seconds * 1000;
+    ms += serializedEntry.timestamp.nanos / 1e6;
+    serializedEntry.timestamp = new Date(ms);
```

Comment:
Forgive me if it's there and I misread it, but do we have a test for this? I'm only seeing tests for `Date` objects being converted to `timestamp` values.